eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -274,24 +278,33 @@ object Router {
 
   // @formatter:off
   case class ChannelDesc(shortChannelId: ShortChannelId, a: PublicKey, b: PublicKey)
-  case class PublicChannel(ann: ChannelAnnouncement, fundingTxid: ByteVector32, capacity: Satoshi, update_1_opt: Option[ChannelUpdate], update_2_opt: Option[ChannelUpdate]) {
+  case class PublicChannel(ann: ChannelAnnouncement, fundingTxid: ByteVector32, capacity: Satoshi, balance_1_opt: Option[MilliSatoshi], update_1_opt: Option[ChannelUpdate], balance_2_opt: Option[MilliSatoshi], update_2_opt: Option[ChannelUpdate]) {
```

Comment:
My gut feeling is that adding those two arguments next to the existing ones doesn't feel right. The `PublicChannel` class holds *verified*, *static* information (expecially `ChannelAnnouncement` and `fundingTxid`, but also `update_X_opt` contain the timestamp at which they are valid). Current balances are pretty different and are more akin to a metadata, they are "always late", not updated the same way, etc. Furthermore, `update_X_opt` are optional because they may not exist, whereas balances always exist, but we just may not know them.

How about regrouping those meta information in a self-contained attribute to make that more clear? The code could be reused for public and private channels too.
```suggestion
  case class PublicChannel(ann: ChannelAnnouncement, fundingTxid: ByteVector32, capacity: Satoshi, update_1_opt: Option[ChannelUpdate], update_2_opt: Option[ChannelUpdate], meta: Option[ChannelMeta]) {
```

We could use this meta to store more data, information about past successes, and so on. The more complex our route finding algo is, the more data it will probably need.

Also, we may only know the balances of a channel for local channels right, which doesn't appear in the model. Not sure if that's a good thing or not.

Reply:
> How about regrouping those meta information in a self-contained attribute to make that more clear?

Good idea, I'll update that, this will allow more extensibility in the future.

> Also, we may only know the balances of a channel for local channels right, which doesn't appear in the model. Not sure if that's a good thing or not.

I hesitated to make `PublicChannel` a trait, and have a `LocalPublicChannel` and `RemotePublicChannel`. But I abandoned that because we could have balance information for non-local channels. If you deploy multiple nodes in the network, it may be (in the far future) useful to synchronize the balances between your nodes to improve your path-finding heuristics. So I prefer the solution of adding a `meta` that may contain balances to all channels, regardless of local/remote.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1382/commits/4a43bd8568434bf32d9f83a0fdf5ad8177b60986